From: =?utf-8?q?Jonas_Dre=C3=9Fler?= <verdre@v0yd.nl>
Date: Fri, 10 Jul 2020 11:05:21 +0000
Subject: tests/conform/actor-pivot-point: Show stage before allocating

We're going to enforce some invariants a bit stricter and will only
allow allocating if an actor is mapped, not only visible.

Since actors are only mapped if their parent is mapped and stages are
hidden by default, we need to show the stage to ensure the actors are
mapped before we allocate them. So do that and call clutter_actor_show()
on the stage before fake-allocating the test actors.

https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1366

(cherry picked from commit 650efb6445e2e76cb3cd47189b8d0d45b1da495e)

Origin: upstream, 3.38.2
---
 src/tests/clutter/conform/actor-pivot-point.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/tests/clutter/conform/actor-pivot-point.c b/src/tests/clutter/conform/actor-pivot-point.c
index fc196b0..38481ba 100644
--- a/src/tests/clutter/conform/actor-pivot-point.c
+++ b/src/tests/clutter/conform/actor-pivot-point.c
@@ -22,6 +22,8 @@ actor_pivot (void)
   clutter_actor_add_child (stage, actor_implicit);
   clutter_actor_add_child (stage, actor_explicit);
 
+  clutter_actor_show (stage);
+
   /* Fake allocation or pivot-point will not have any effect */
   clutter_actor_allocate (actor_implicit, &allocation);
   clutter_actor_allocate (actor_explicit, &allocation);
